lib/vmpooler/api/v1.rb

```diff
@@ -245,36 +307,85 @@ def sync_pool_templates
       pool_index = pool_index(pools)
       template_configs = backend.hgetall('vmpooler__config__template')
       template_configs&.each do |poolname, template|
-          if pool_index.include? poolname
-            unless pools[pool_index[poolname]]['template'] == template
-              pools[pool_index[poolname]]['template'] = template
-            end
-          end
+        next unless pool_index.include? poolname
+
+        pools[pool_index[poolname]]['template'] = template unless pools[pool_index[poolname]]['template'] == template
```

Comment:
What's the original logic behind this? Is doing an assignment to this hash expensive somehow?

Reply:
The configured template in redis may be different than the template configured in the configuration file on disk for vmpooler because the value provided via the configuration API takes precedence. This was added to ensure that the API got the expected data in the configuration at startup time.

Reply:
But wouldn't just assigning the template accomplish the same thing? I guess I'm not understanding why the conditional checking that the values aren't the same is necessary, as if `pools[pool_index[poolname]]['template'] == template` then the assignment is a no-op (maybe this is my misunderstanding), and if `pools[pool_index[poolname]]['template'] != template` then assignment is fine.

Reply:
Yes, I think it would. I have adjusted it. Let me know if I misunderstood your suggestion.

Reply:
Yep, the changes in https://github.com/puppetlabs/vmpooler/pull/375/commits/f596a989a5acbe187ed0c905e24080939a826994 make way more sense to me. :)